express-notes-api/server.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import express from 'express';
+import { writeFile, readFile } from 'node:fs/promises';
+
+const app = express();
+app.use(express.json());
+
+const json = await readFile('data.json', 'utf8'); // string
```

Comment:
When you run in express the server stays running, so global variables like this can get out of sync with the file system. It's safest to always read the file at the beginning of every route, then write it at the end.

When we get to databases you will be forced to read the data in every route.